integration_tests/src/main/python/array_test.py

```diff
@@ -449,7 +449,7 @@ def q1(spark):
 
 @incompat
 @pytest.mark.parametrize('data_gen', no_neg_zero_all_basic_gens + decimal_gens, ids=idfn)
-@pytest.mark.skipif(is_before_spark_313() or is_spark_330(), reason="NaN equality is only handled in Spark 3.1.3+")
+@pytest.mark.skipif(is_before_spark_313() or is_spark_330() or is_spark_330cdh(), reason="NaN equality is only handled in Spark 3.1.3+")
```

Comment:
"only handled in Spark 3.1.3+" so that also work with Spark 3.3.0, doesn't it? Why 3.3.0 is special here? If the bug occurs again with that version, please update the comment.

Reply:
There was a different bug in Spark 3.3.0. I will update the comment accordingly.